packages/mdc-ripple/_mixins.scss

```diff
@@ -201,38 +214,51 @@
   }
 }
 
-@mixin mdc-states-hover-opacity($opacity, $query: mdc-feature-all()) {
+@mixin mdc-states-hover-opacity(
+  $opacity, $query: mdc-feature-all(), $ripple-target: null) {
   $feat-color: mdc-feature-create-target($query, color);
 
+  $ripple-target-selector: if($ripple-target == null, "&", " #{$ripple-target}");
+
   // Background wash styles, for both CSS-only and upgraded stateful surfaces
-  &:hover::before {
-    // Opacity falls under color because the chosen opacity is color-dependent in typical usage
-    @include mdc-feature-targets($feat-color) {
-      opacity: $opacity;
+  &:hover {
+    #{$ripple-target-selector}::before {
+      // Opacity falls under color because the chosen opacity is color-dependent in typical usage
+      @include mdc-feature-targets($feat-color) {
+        opacity: $opacity;
+      }
     }
   }
 }
 
-@mixin mdc-states-focus-opacity($opacity, $has-nested-focusable-element: false, $query: mdc-feature-all()) {
+@mixin mdc-states-focus-opacity(
+  $opacity,
+  $has-nested-focusable-element: false,
+  $query: mdc-feature-all(),
+  $ripple-target: null) {
   $feat-animation: mdc-feature-create-target($query, animation);
   // Opacity falls under color because the chosen opacity is color-dependent in typical usage
   $feat-color: mdc-feature-create-target($query, color);
 
+  $ripple-target-selector: if($ripple-target == null, "", " #{$ripple-target}");
+
   // Focus overrides hover by reusing the ::before pseudo-element.
   // :focus-within generally works on non-MS browsers and matches when a *child* of the element has focus.
   // It is useful for cases where a component has a focusable element within the root node, e.g. text field,
   // but undesirable in general in case of nested stateful components.
   // We use a modifier class for JS-enabled surfaces to support all use cases in all browsers.
   $cssOnlyFocusSelector: if(
     $has-nested-focusable-element,
-    "&:not(.mdc-ripple-upgraded):focus::before, &:not(.mdc-ripple-upgraded):focus-within::before",
-    "&:not(.mdc-ripple-upgraded):focus::before"
+    "&:not(.mdc-ripple-upgraded):focus#{$ripple-target-selector}::before," +
```

Comment:
We can refactor this to use nested block selectors here too. :) 

Example: https://www.sassmeister.com/gist/58a8f62bec24b96f4c13f1a5f34a4d5b